### PR TITLE
ci: fix rust build target settings in CI

### PR DESCRIPTION
**Pull request by eitsupi**
Follow up #2790

### CODE REVIEW

File: .github/workflows/test-rust.yaml [hunk 1]
```diff
@@ -22,7 +22,7 @@ jobs:
     steps:
       - name: 📂 Checkout code
         uses: actions/checkout@v3
-      - if: inputs.os == 'Linux'
+      - if: ${{ contains(inputs.target, 'musl') }}
```
Comment (max-sixty) [inline .github/workflows/test-rust.yaml:25]: 👍

**Review by max-sixty — APPROVED** [review]

**Comment by eitsupi** [comment]
Sorry, Prettier doesn't work with my VS Code and I can't format the yaml file.

**Comment by max-sixty** [comment]
It's weird that it now just works! Sorry for the change, it seems like it wasn't actually a problem.

**Comment by eitsupi** [comment]
I have reverted to musl only for prqlc builds. The tests are still on gnu.

As you know, the default target on Linux is gnu, and this seems to be because musl requires additional tools to be installed, etc.? (you should be able to find various articles on this if you look it up)

**Comment by max-sixty** [comment]
Ah that explains it! 

That doesn't share the cache, so this undoes some of the work. But the ubuntu builds are fast anyway, so it only takes the total time from 2.5 to 3.5 minutes. If you want to try and get the tests working on musl then that could be good. But overall we've done a lot here and it's in a good state, we can leave it for a while.

Thank you!

**Comment by max-sixty** [comment]
(and FYI I downloaded it and tested it — it works. Kinda weird the tests don't work then; I'm fairly confident I had added the `apt-get` command)

**Pull request merged**